spec/attributes/type/stripped_string_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'rails_helper'
+
+RSpec.describe Type::StrippedString do
+  subject { described_class.new }
+
+  let(:coerced_value) { subject.cast(value) }
+
+  describe 'registry' do
+    it 'is registered with type `:string`' do
+      expect(
+        ActiveModel::Type.lookup(:string).is_a?(described_class)
+      ).to eq(true)
+    end
+
+    it 'has an underlying type of `:string`' do
+      expect(subject.type).to eq(:string)
+    end
+  end
+
+  describe 'when value is `nil`' do
+    let(:value) { nil }
+    it { expect(coerced_value).to be_nil }
+  end
+
+  describe 'when value is a symbol' do
+    let(:value) { :foobar }
+    it { expect(coerced_value).to eq('foobar') }
+  end
+
+  describe 'when value is a string' do
+    let(:value) { 'foobar' }
+    it { expect(coerced_value).to eq('foobar') }
+  end
+
+  describe 'strip spaces from the string value' do
+    context 'leading or trailing spaces' do
+      let(:value) { ' foobar ' }
+      it { expect(coerced_value).to eq('foobar') }
+    end
+
+    context 'more than one leading or trailing spaces' do
+      let(:value) { '   foobar   ' }
+      it { expect(coerced_value).to eq('foobar') }
+    end
+
+    describe 'intermediate spaces' do
+      let(:value) { ' foo bar ' }
+      it { expect(coerced_value).to eq('foo bar') }
+    end
+
+    describe 'more than one intermediate spaces' do
+      let(:value) { ' foo   bar ' }
+      it { expect(coerced_value).to eq('foo   bar') }
+    end
```

Comment:
Small thing but I guess we could test 'foobar   ' and '   foobar' 🤔

Reply:
Yep, as I know internally it uses `#strip` method, it will do lead and trail, but you're right an external observer might not know the specific implementation so a explicit expectation for only lead blank / trail blanks will not hurt, will add it 👍